File: moderate/pascals_triangle.rb
File.open(ARGV[0]).each_line do |line|
  print 1
  (1...line.to_i).each do |i|
    print ' 1'
    r = 1
    i.times do |j|
      r = (r * (i - j)) / (j + 1)
      print ' ' + r.to_s
    end
  end
  puts
end

File: moderate/pascals_triangle.pl
use strict;
open( INFILE, "<", $ARGV[0] )
  or die("Cannot open file $ARGV[0] for reading: $!");
while ( my $line = <INFILE> ) {
    print "1";
    for ( my $i = 1 ; $i < $line ; $i++ ) {
        print " 1";
        my $r = 1;
        for ( my $j = 0 ; $j < $i ; $j++ ) {
            $r = ( $r * ( $i - $j ) ) / ( $j + 1 );
            print " $r";
        }
    }
    print "\n";
}
close(INFILE);
